Make changes a deque

eseliger opened the pull request:
That will give us changelist.Clear for free through destructor calls at the end of the processing loop iteration in the main thread.

## Test plan

CI.

eseliger committed:
Make changes a deque

That will give us changelist.Clear for free through destructor calls at the end of the processing loop iteration in the main thread.

ggilmore commented on p4-fusion/main.cc:
```diff
@@ -218,14 +221,22 @@ int Main(int argc, char** argv)
+		// Now move the changelist and pop it off the queue.
+		// Once this iteration is over, it will be destructed
+		// and memory is freed.
+		ChangeList cl = std::move(changes.front());
```
Not an actionable comment, but I found it to be an unfortunate subtlety that we need to wait explicitly for the workers to stop referencing the old memory address before we use std::move here.

eseliger replied:
yes it's quite annoying - I think a rework using two queues could fix it where we only schedule changelists for processing once the download is done but I'll save that for a future one

eseliger committed:
Cleanup log output (#24)

The 1.12 version of p4-fusion was less verbose which had the nice property that since we always show the last log line in the clone progress UI it would contains stats on the progress.
This reverts to that, and makes the output required for our test script opt-in using a compile time flag.

eseliger committed:
Revert "make changedfilegroups a local variable"

This reverts commit 8a85d41ff0a0710950259fe2d2d5278e42804e17.
